### PR TITLE
[SL Alpah5] Add initial implementation of the connector

### PR DESCRIPTION
**Pull request by sachinira**
Resolves https://github.com/wso2-enterprise/choreo/issues/5310
Resolves https://github.com/wso2-enterprise/choreo/issues/5311

**Design document** - https://docs.google.com/document/d/1qKgg4fSlKyOkLR7KDdAjz7xB0aXvYXTIIs0ojRw8fQE/edit?usp=sharing

## Automation tests
 - Unit tests 
   > All the operations are tested separately
 - Integration tests
   > All the operations are tested as a flow

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

https://github.com/sachinira/module-ballerinax-microsoft.teams/tree/teams/teams/samples

 
https://docs.microsoft.com/en-us/graph/api/resources/teams-api-overview?view=graph-rest-1.0
https://docs.microsoft.com/en-us/graph/teams-concept-overview
> MS Teams connector in Logic Apps
https://docs.microsoft.com/en-us/connectors/teams/

### CODE REVIEW

File: teams/data_mappings.bal [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ChannelData[] channelArray = [];
+    json[] valueArrayResult = <json[]>reponsePayload.get("value");
+
+    foreach var item in valueArrayResult {
+        ChannelData channel = check item.cloneWithType(ChannelData);
+        channelArray.push(channel);
+    }
+    return channelArray;
+}
+
```
Comment (Chuhaa) [inline teams/data_mappings.bal:27]:
```suggestion
isolated function mapJsonToChannelDataArray(map<json> reponsePayload) returns ChannelData[]|Error {
     return check reponsePayload["value"].cloneWithType(ChannelDataArray);
}

type ChannelDataArray ChannelData[];
```

File: teams/samples/add_member_to_channel.bal [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// under the License.
+
+import ballerina/log;
+import ballerina/os;
+import ballerinax/microsoft.teams;
+
+configurable string & readonly refreshUrl = os:getEnv("REFRESH_URL");
+configurable string & readonly refreshToken = os:getEnv("REFRESH_TOKEN");
+configurable string & readonly clientId = os:getEnv("CLIENT_ID");
+configurable string & readonly clientSecret = os:getEnv("CLIENT_SECRET");
```
Comment (Chuhaa) [inline teams/samples/add_member_to_channel.bal:24]: We can remove `os:getEnv` in samples. This is not required if a user wants access from `Cofig.toml`
`configurable string & readonly refreshUrl = ?`

Comment (sachinira) [reply teams/samples/add_member_to_channel.bal]: Fixed this thanks

File: teams/data_mappings.bal (hunk 1 above)
Comment (sachinira) [reply teams/data_mappings.bal]: Fixed this thanks for the suggestion

File: teams/Dependancies.toml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+[[dependency]]
```
Comment (LakshanSS) [inline teams/Dependancies.toml:1]: Check the name of this file.
There is a typo.

It should be `Dependencies.toml`

Comment (sachinira) [reply teams/Dependancies.toml]: Fixed this thanks

**Review by abeykoon — APPROVED** [review]

**Comment by sachinira** [comment]
Fixed by https://github.com/ballerina-platform/module-ballerinax-microsoft.teams/pull/3

**Pull request closed without merging**